index.js

```diff
@@ -117,7 +118,7 @@ function commonJSLookup(partial, filename, directory) {
   var result = '';
 
   try {
-    result = require.resolve(partial);
+    result = resolve.sync(partial, {basedir: path.dirname(filename)});
```

Comment:
I believe the assumption for the original implementation was that `filename` was within `directory`. I'm curious, what is the value of `require.main.paths` before and after your fix?